include/revng/Model/ABI.h

```diff
@@ -6,24 +6,319 @@
+  SystemZ_s390x,
+
+  Count
+};
+
+inline constexpr llvm::StringRef getName(Values V) {
   switch (V) {
   case Invalid:
     return "Invalid";
+
   case SystemV_x86_64:
     return "SystemV_x86_64";
+  case SystemV_x86:
+    return "SystemV_x86";
+  case SystemV_x86_regparm_3:
+    return "SystemV_x86_regparm_3";
+  case SystemV_x86_regparm_2:
+    return "SystemV_x86_regparm_2";
+  case SystemV_x86_regparm_1:
+    return "SystemV_x86_regparm_1";
+
+  case Microsoft_x86_64:
+    return "Microsoft_x86_64";
+  case Microsoft_x86_64_vectorcall:
+    return "Microsoft_x86_64_vectorcall";
+  case Microsoft_x86_64_clrcall:
+    return "Microsoft_x86_64_clrcall";
+
+  case Microsoft_x86_cdecl:
+    return "Microsoft_x86_cdecl";
+  case Microsoft_x86_stdcall:
+    return "Microsoft_x86_stdcall";
+  case Microsoft_x86_thiscall:
+    return "Microsoft_x86_thiscall";
+  case Microsoft_x86_fastcall:
+    return "Microsoft_x86_fastcall";
+  case Microsoft_x86_clrcall:
+    return "Microsoft_x86_clrcall";
+  case Microsoft_x86_vectorcall:
+    return "Microsoft_x86_vectorcall";
+
+  case Pascal_x86:
+    return "Pascal_x86";
+
+  case AArch64:
+    return "AArch64";
+  case ARM:
+    return "ARM";
+
+  case SystemV_MIPS_o32:
+    return "SystemV_MIPS_o32";
+
+  case SystemZ_s390x:
+    return "SystemZ_s390x";
+
+  default:
+    revng_abort();
+  }
+}
+
+inline constexpr model::Architecture::Values getArchitecture(Values V) {
+  switch (V) {
+  case SystemV_x86_64:
+  case Microsoft_x86_64:
+  case Microsoft_x86_64_clrcall:
+  case Microsoft_x86_64_vectorcall:
+    return model::Architecture::x86_64;
+
+  case SystemV_x86:
+  case SystemV_x86_regparm_3:
+  case SystemV_x86_regparm_2:
+  case SystemV_x86_regparm_1:
+  case Microsoft_x86_clrcall:
+  case Microsoft_x86_vectorcall:
+  case Microsoft_x86_cdecl:
+  case Microsoft_x86_stdcall:
+  case Microsoft_x86_fastcall:
+  case Microsoft_x86_thiscall:
+  case Pascal_x86:
+    return model::Architecture::x86;
+
+  case AArch64:
+    return model::Architecture::aarch64;
+  case ARM:
+    return model::Architecture::arm;
+
+  case SystemV_MIPS_o32:
+    return model::Architecture::mips;
+
+  case SystemZ_s390x:
+    return model::Architecture::systemz;
+
+  case Count:
+  case Invalid:
   default:
     revng_abort();
   }
-  revng_abort();
+}
+
+inline constexpr model::abi::Values getDefault(model::Architecture::Values V) {
+  switch (V) {
+  case model::Architecture::x86:
+    return SystemV_x86;
+  case model::Architecture::arm:
+    return ARM;
+  case model::Architecture::mips:
+    return SystemV_MIPS_o32;
+  case model::Architecture::x86_64:
+    return SystemV_x86_64;
+  case model::Architecture::aarch64:
+    return AArch64;
+  case model::Architecture::systemz:
+    return SystemZ_s390x;
+
+  case model::Architecture::Invalid:
+  case model::Architecture::Count:
+  default:
+    revng_abort();
+  }
+}
+
+inline constexpr std::string_view getDescription(model::abi::Values V) {
```

Comment:
For the sake of coherence let's use `StringRef`.

Reply:
[Done](https://github.com/revng/revng/pull/98/commits/faaad6111cc9ad4bb68c78622aa114c7031435ec).